src/main/java/net/snowflake/client/core/SFSessionProperty.java

```diff
@@ -85,7 +85,16 @@ public enum SFSessionProperty {
   DISABLE_GCS_DEFAULT_CREDENTIALS("disableGcsDefaultCredentials", false, Boolean.class),
 
   JDBC_ARROW_TREAT_DECIMAL_AS_INT("JDBC_ARROW_TREAT_DECIMAL_AS_INT", false, Boolean.class),
-  DISABLE_SAML_URL_CHECK("disableSamlURLCheck", false, Boolean.class);
+
+  DISABLE_SAML_URL_CHECK("disableSamlURLCheck", false, Boolean.class),
+
+  // Used to determine whether a hardcoded value should be used for the formatter (for backwards
+  // compatibility)
+  JDBC_DEFAULT_FORMAT_DATE_WITH_TIMEZONE(
+      "JDBC_DEFAULT_FORMAT_DATE_WITH_TIMEZONE", false, Boolean.class),
+
+  // Used as a fix for issue 244. Remove with snowflake-jdbc version 4.x with BCR changes.
```

Comment:
@sfc-gh-dprzybysz do we have a tag or anything on which we will be looking for things to change in 4.x?

Reply:
we will have a task for date time types analyzis in JDBC
for upgrade to 4.x we have indeed SNOW-1050477